plugins/woocommerce/src/Internal/Admin/Marketing.php

```diff
@@ -140,6 +157,74 @@ protected function register_overview_page() {
 		}
 	}
 
+	/**
+	 * Order marketing menu items alphabeticaly.
+	 * Overview should be first, and Coupons should be second, followed by other marketing menu items.
+	 *
+	 * @return  void
+	 */
+	public function reorder_marketing_submenu() {
+		global $submenu;
+
+		if ( ! isset( $submenu['woocommerce-marketing'] ) ) {
+			return;
+		}
+
+		$marketing_submenu = $submenu['woocommerce-marketing'];
+		$new_menu_order    = array();
+
+		// Overview should be first.
+		$overview_key = array_search( 'Overview', array_column( $marketing_submenu, self::SUBMENU_NAME_KEY ), true );
+		if ( false !== $overview_key ) {
+			$new_menu_order[] = $marketing_submenu[ $overview_key ];
+			array_splice( $marketing_submenu, $overview_key, 1 );
+		}
+
+		if ( false === $overview_key ) {
+			/*
+			 * If Overview is not found we may be on a site witha different language.
+			 * We can use a fallback and try to find the overview page by its path.
+			 */
+			$overview_key = array_search( 'admin.php?page=wc-admin&path=/marketing', array_column( $marketing_submenu, self::SUBMENU_LOCATION_KEY ), true );
+			if ( false !== $overview_key ) {
+				$new_menu_order[] = $marketing_submenu[ $overview_key ];
+				array_splice( $marketing_submenu, $overview_key, 1 );
+			}
+		}
```

Comment:
> Actually, since we are searching for a location key, it feels like searching by name key is redundant and not necessary.

I don't trust that the location can be the same on all of the setups. I suspect that it may be different in the future. Granted it should be adjusted here in such a case but I want to be defensive. By searching for the name first, I think that we cover more cases.

Reply:
The code has been refactored.